src/aktualizr_repo/director_repo.cc

```diff
@@ -17,6 +17,11 @@ void DirectorRepo::addTarget(const std::string &target_name, const Json::Value &
   director_targets["targets"][target_name] = target;
   director_targets["targets"][target_name]["custom"].removeMember("hardwareIds");
   director_targets["targets"][target_name]["custom"]["ecuIdentifiers"][ecu_serial]["hardwareId"] = hardware_id;
+  if (!url.empty()) {
+    director_targets["targets"][target_name]["custom"]["uri"] = url;
+  } else {
+    director_targets["targets"][target_name]["custom"].removeMember("uri");
```

Comment:
Why a nested structure is required for the custom URI, not just a single attribute of `target_name`, to paraphrase why ["custom"]["uri"] not just ["custom-uri"] or there is/are other attribute(s) of ["custom"] ?
Perhaps, just the whole "custom" can be removed ?

Reply:
> there is/are other attribute(s) of ["custom"] ?

Precisely. It's following the recommendation of Uptane to keep implementation-specific custom metadata in a `custom` section.